### ISSUE TITLE
array_merge don't return list type

### ISSUE THREAD

**Issue by vadim2991** [vimeo/psalm#2642]
If use array_merge with list arguments return type should be list
https://psalm.dev/r/937292dfcd
```php
<?php
/**
* @param list<int> $foo
*/
function foo(array $foo) : void {
    var_export($foo);
}

$foo1 = [1, 2, 3];
$foo2 = [1, 4, 5];
$fooMerged = array_merge($foo1, $foo2);

foo($foo1);
foo($foo2);
foo($fooMerged);
```
Actual: ERROR: InvalidArgument - 15:5 - Argument 1 of foo expects list<int>, array{0: int(1), 1: int(2), 2: int(3), 3: int(1), 4: int(4), 5: int(5)} provided
Expected: no errors

**Comment by SignpostMarv** [comment]
works if manually declared as `list<int>`: https://psalm.dev/r/d5c0079603
